pkgs/test_api/lib/src/backend/compiler.dart

```diff
@@ -0,0 +1,51 @@
+// Copyright (c) 2023, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// All the Dart compilers supported by the test runner.
+class Compiler {
+  /// The production Dart to Javascript compiler (whole world, optimizing).
+  static const Compiler dart2js =
+      Compiler('Dart2Js', 'dart2js', isDefault: true);
+
+  /// Experimental Dart to Wasm compiler.
+  static const Compiler dart2wasm = Compiler('Dart2WASM', 'dart2wasm');
+
+  /// The standard compiler for vm tests, compiles tests to kernel before
+  /// running them on the VM.
+  static const Compiler kernel = Compiler('Kernel', 'kernel', isDefault: true);
+
+  /// Runs tests directly from source, with no precompilation.
+  static const Compiler source = Compiler('Source', 'source');
+
+  /// The compilers that are supported by the test runner by default.
+  static const List<Compiler> builtIn = [
+    Compiler.dart2js,
+    Compiler.dart2wasm,
+    Compiler.kernel,
+    Compiler.source,
+  ];
+
+  /// The human-friendly name of the compiler.
+  final String name;
+
+  /// The identifier used to look up the compiler.
+  final String identifier;
+
+  /// Whether this compiler is enabled by default.
+  final bool isDefault;
+
+  const Compiler(this.name, this.identifier, {this.isDefault = false});
+
+  /// Converts a JSON-safe representation generated by [serialize] back into a
```

Comment:
[nit] mention that only the `builtIn` compilers are supported.

We could consider supporting serialization and deserialization to a `Map` for unrecognized compilers, but it's probably not necessary at the moment, since no other part of the system will support pluggable unrecognized compilers.

If and when we get around to looking at proper pluggable platform support, it would be nice to make compiler pluggable too.

Reply:
Yeah I figured we would only add support for custom compilers if requested/needed later on.

Reply:
I went ahead and just removed the public constructor.